mlx/directives/item_matrix_directive.py

```diff
@@ -99,17 +100,33 @@ def perform_replacement(self, app, collection):
             covered = False
             rights = [[] for _ in range(int(bool(self['intermediate'])) + len(self['target']))]
             if mapping_via_intermediate:
-                covered = source_id in mapping_via_intermediate
-                if covered:
-                    args = [rows, source_item, rights, app]
-                    duplicate_source_count += self._store_source_via_intermediate(mapping_via_intermediate[source_id],
-                                                                                  *args)
+                if not mapping_via_intermediate[source_id]:
+                    self._store_data(rows, source_item, rights, covered, app)
+                    continue
+                covered_intermediates = {intermediate: any(target_set for target_set in target_sets)
+                                         for intermediate, target_sets in mapping_via_intermediate[source_id].items()}
+
+                if self['coveredintermediates']:
+                    covered = all(covered_intermediates.values())
+                else:
+                    covered = any(covered_intermediates.values())
+
+                if self['splitintermediates']:
+                    for intermediate, target_sets in mapping_via_intermediate[source_id].items():
+                        self._store_row_with_intermediate({intermediate: target_sets},
+                                                            rows, source_item, rights, covered, app)
```

Comment:
```suggestion
                    for intermediate, target_sets in intermediates.items():
                        self._store_row_with_intermediate({intermediate: target_sets},
                                                           rows, source_item, rights, covered, app)
```
This suggestion depends on my first suggestion.